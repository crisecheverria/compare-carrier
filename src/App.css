.media {
    align-items: center;
    display: flex;
    text-align: left;
}

p {
    margin-bottom: 0.1em !important;
}

.custom-box {
    background-color: white;
    border-radius: 16px;
    border-style: solid;
    border-color: hsl(0, 0%, 95%);
    color: #4a4a4a;
    display: block;
    padding: 1.25rem;
}

.is-active {
    background-color: hsl(0, 0%, 83%);
    border-color: transparent !important;
}

.progress {
    -moz-appearance: none;
    -webkit-appearance: none;
    border: none;
    border-radius: 290486px;
    display: inline-block;
    height: 1rem;
    overflow: hidden;
    padding: 0;
    margin-left: 15px;
    width: 70%;
}

.currency {
    font-size: 1.2em;
}