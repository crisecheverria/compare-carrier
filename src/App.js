import React, { useState } from "react"
import Buttons from "./components/Buttons"
import Products from "./components/Products"
import "./App.css"

function App() {
  const [parameterState, setParameterState] = useState("")

  function sortByParameter(parameter) {
    setParameterState(parameter)
  }

  return (
    <section className="hero is-fullheight">
      <div className="hero-body">
        <div className="container">
          <h2>
            <strong>Sort by</strong>
          </h2>
          <Buttons sortBy={sortByParameter} />
          <Products sortParameter={parameterState} />
        </div>
      </div>
    </section>
  )
}

export default App
