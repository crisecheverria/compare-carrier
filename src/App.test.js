import React from "react"
import { render } from "@testing-library/react"
import App from "./App"

test("renders <App /> component", () => {
  const { getByText, getByTestId, baseElement } = render(<App />)
  const sortingTitle = getByText(/sort by/i)
  const buttonCheapest = getByTestId(/btn-cheapest/i)
  const buttonFastest = getByTestId(/btn-fastest/i)
  // const columnsProducts = getByTestId(/products/i)

  expect(sortingTitle).toBeInTheDocument()
  expect(buttonCheapest).toBeInTheDocument()
  expect(buttonFastest).toBeInTheDocument()
  expect(baseElement).toMatchSnapshot()
  // expect(columnsProducts).toBeInTheDocument()
})
