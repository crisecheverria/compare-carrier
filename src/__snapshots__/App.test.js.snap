// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> component 1`] = `
<body>
  <div>
    <section
      class="hero is-fullheight"
    >
      <div
        class="hero-body"
      >
        <div
          class="container"
        >
          <h2>
            <strong>
              Sort by
            </strong>
          </h2>
          <h2
            class="subtitle"
          >
            <button
              class="button is-light"
              data-testid="sorting-options"
              name="cheapest"
              type="button"
            >
              cheapest
            </button>
            <button
              class="button is-light"
              data-testid="sorting-options"
              name="fastest"
              type="button"
            >
              fastest
            </button>
          </h2>
          <div
            class="columns is-desktop"
          >
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="DHL 1-logo"
                        class="is-rounded"
                        src="images/dhl.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        DHL 1
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          62.98
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          2 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="2"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="UPS 1-logo"
                        class="is-rounded"
                        src="images/ups.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        UPS 1
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          134.00
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          2 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="2"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="DHL 2-logo"
                        class="is-rounded"
                        src="images/dhl.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        DHL 2
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          147.23
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          2 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="2"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
          <div
            class="columns is-desktop"
          >
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="TNT 1-logo"
                        class="is-rounded"
                        src="images/tnt.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        TNT 1
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          416.21
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          1 day
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="1"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="TNT 2-logo"
                        class="is-rounded"
                        src="images/tnt.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        TNT 2
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          98.23
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          3 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="3"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="TNT 3-logo"
                        class="is-rounded"
                        src="images/tnt.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        TNT 3
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          349.21
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          1 day
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="1"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
          <div
            class="columns is-desktop"
          >
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="DHL 3-logo"
                        class="is-rounded"
                        src="images/dhl.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        DHL 3
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          1342.76
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          1 day
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="1"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="UPS 2-logo"
                        class="is-rounded"
                        src="images/ups.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        UPS 2
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          123.00
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          2 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="2"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="column"
              data-testid="products"
            >
              <div
                class="custom-box"
              >
                <article
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-64x64"
                    >
                      <img
                        alt="DSV 1-logo"
                        class="is-rounded"
                        src="images/dsv.svg"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content"
                  >
                    <div
                      class="content"
                    >
                      <p
                        data-testid="carrier"
                      >
                        DSV 1
                      </p>
                      <p
                        class="currency"
                        data-testid="cost"
                      >
                        <strong>
                          176.00
                        </strong>
                      </p>
                      <span>
                        <strong
                          data-testid="time"
                        >
                          2 days
                        </strong>
                        <progress
                          class="progress is-warning"
                          max="7"
                          value="2"
                        />
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
